pandas/core/series.py

```diff
@@ -2163,12 +2197,30 @@ def idxmax(self, axis=0, skipna=True, *args, **kwargs):
 
         >>> s.idxmax(skipna=False)
         nan
+
+        >>> s.idxmax(keep='all')
+        array(['C', 'E'], dtype=object)
+
+        >>> s.idxmax(keep='last')
+        'E'
         """
         skipna = nv.validate_argmax_with_skipna(skipna, args, kwargs)
-        i = nanops.nanargmax(self._values, skipna=skipna)
+        if keep == "last":
```

Comment:
rather than repeating all of this code in Series & DataFrame; i'd like to push it down to algorithms.py